### ISSUE TITLE
Field filtering misses some fields

### ISSUE THREAD

**Issue by rhuss** [atlasmap/atlasmap-ui#94]
@kcbabo : 

>  When the data mapper loads the Twitter Status object, it does not include all fields of the Status model.  The user.screenName field would be an example.

> What's crazy is that when it first opens up, the model viewer shows a count of 89 fields at the bottom for the input model.  If you click on the user field within the Status object, it then goes to 141 and user.screenName is then visible.  

@rhuss:

> I experienced the same problem. However, if you first navigate through the tree (not by filtering), the user has the `screenName`. After this it is also available by the filter search. It's not really clear to me when and how this happens.

**Comment by apupier** [comment]
maybe worth checking if it is not the return of this old issue for which I observed the number of fields increasing also: https://github.com/syndesisio/syndesis-ui/issues/680

**Comment by igarashitm** [comment]
Sounds like same root cause with #83 , lazy loading of subsequent model DOM

**Comment by igarashitm** [comment]
@jpav , can you look into this one? It looked like you got a deeper idea for this issue after talked with Matt :)

**Comment by jpav** [comment]
Yes, will do

**Comment by jpav** [comment]
As @igarashitm pointed out, this is basically a duplicate of #83.  One thing that I forgot @mattrpav mentioned in this issue is part of the problem is related to the infinite recursion built into the twitter model, requiring code to limit the depth that the model is analyzed.  However, even with that, the user name should show up somewhere within the search results, just not necessarily at the depth expected.  In any case with infinite recursion in the model, we won't be able to show a true count of the objects within the model. We may have to come up with some way in the UI to indicate that recursion is present at certain levels so the user knows not to expect search results at arbitrary depths.

**Comment by kcbabo** [comment]
Yeah, I think the main issue is search not returning second-level fields (user.screenName), which is a bit of a bummer.  Total field count is probably a bit pedantic for most users.

**Comment by igarashitm** [comment]
I'd +1 removing field count for searching. I haven't yet looked into the details though, so many Java classes have cyclic references, which easily causes infinite depth.
